plugins/infinispan90/src/main/java/org/radargun/service/JGroups4Dumper.java

```diff
@@ -0,0 +1,58 @@
+package org.radargun.service;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.concurrent.ThreadPoolExecutor;
+import org.jgroups.protocols.TP;
+import org.jgroups.stack.Protocol;
+import org.jgroups.stack.ProtocolStack;
+import org.jgroups.util.TimeScheduler;
+import org.radargun.utils.Utils;
+
+/**
+ * JGroupsDumper modified for JGroups 4
+ *
+ * @author Roman Macor (rmacor@redhat.com)
+ */
+public class JGroups4Dumper extends JGroupsDumper {
+
+   public JGroups4Dumper(ProtocolStack protocols, long interval) {
+      super(protocols, interval);
+   }
+
+   @Override
+   protected void addDumpers(){
+      dumpers.add(new UNICAST3Dumper());
+      dumpers.add(new JGroups4Dumper.TPDumper());
+   }
+
+   @Override
+   protected Map<String, Object> getStats(Protocol protocol){
```

Comment:
you are right, good point, how about this:
return stack.dumpStats(protocol.getName(), null);